### ISSUE TITLE
Better error message for duplicate constraints in input query

### ISSUE THREAD

**Issue by lukashornych** [FgForrest/evitaDB#407]
When certain constraints that are expected to be present in a single query only once are duplicated, evitaDB throws internal error:
```
query(
  collection("Product"),
  require(
    entityFetch(
      priceContentAll(),
      priceContentRespectingFilter()
    )
  )
)
```
This is because we internally use the `QueryUtils.findConstraint(...)` method which throws internal error if multiple constraints are found.

I think these errors should throw end-developer errors so that the developer knows that the query is wrong.

**Comment by novoj** [comment]
Closing as duplicate of #24 (moved there).
